code/weapon/weapons.cpp

```diff
@@ -1361,6 +1361,8 @@ int parse_weapon(int subtype, bool replace, const char *filename)
 		}
 	}
 
+	std::optional<modular_curves_entry> damage_mult_curve;
```

Comment:
As currently written, this won't work.  Consider a modular -wep.tbm that adds a damage multiplier curve and nothing else.  If the weapon is a beam, it won't be applied correctly because the TBM won't contain `$BeamInfo:`.

The best place for applying the curve is probably right after the `$BeamInfo:` parsing, but outside the if() block rather than inside it.  There should be a simple if() check that either checks the `Weapon_Info::Beam` flag or checks `wip->is_beam()`, similar to the original code in this PR that checked `subtype == WP_BEAM`.  Doing everything this way will ensure that the curve will work whether `$BeamInfo:` is parsed or not, and whether the weapon is a beam or not.

Reply:
All right, this good?

Reply:
Yup, looks good!